vernac/vernacentries.ml

```diff
@@ -2093,9 +2093,11 @@ let vernac_declare_reduction ~local s r =
 let vernac_global_check c =
   let env = Global.env() in
   let sigma = Evd.from_env env in
-  let c,uctx = Constrintern.interp_constr env sigma c in
+  let c = Constrintern.intern_constr env sigma c in
+  let sigma, c = Pretyping.understand_tcc ~flags:Pretyping.all_and_fail_flags env sigma c in
```

Comment:
We could also do
```suggestion
  let c,uctx = Constrintern.interp_constr env sigma c in
  let sigma = Evd.from_ctx uctx in
```

Reply:
I would rather not fiddle with evarmaps, as reconstructing them might be not exactly the same as just passing them around. Maybe it is now, but it might change in the future and we don't lose anything by just threading it.